### PR TITLE
Stage 3 review feedback

### LINKED ISSUE #40: Does undefined alias to auto include compact notation?
Right now, this proposal specifies that `undefined` aliases to `"auto"`.
Does this also include `"compact"` notation, considering ICU defaults it to `"min2"` for compact notation so that `true` equates to `'min2"` for compact notation in the current Intl.NumberFormat?

### LINKED ISSUE #42: property name of "TrailingZeroDisplay" should be "trailingZeroDisplay" and the question about the default
https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/diff.html#sec-initializenumberformat
Let trailingZeroDisplay be ? GetOption(options, "TrailingZeroDisplay", "string", « "auto", "stripIfInteger" », "stripIfInteger").

should be
Let trailingZeroDisplay be ? GetOption(options, "trailingZeroDisplay", "string", « "auto", "stripIfInteger" », "stripIfInteger").

also, is it intentional to switch the default to "stripIfInteger"? I cannot find the discussion of such change in  https://github.com/tc39/proposal-intl-numberformat-v3/issues/8 nor in https://github.com/tc39/proposal-intl-numberformat-v3

### LINKED ISSUE #44: Add morePrecision and  lessPrecision RoundingType to "1.5 Properties of Intl.PluralRules Instances"
https://tc39.es/proposal-intl-numberformat-v3/out/pluralrules/diff.html#sec-properties-of-intl-pluralrules-instances

Since InitializePluralRules calls SetNumberFormatDigitOptions it is possible RoundingType in the return value contains morePrecision or  lessPrecision

Change
```
[[RoundingType]] is one of the values fractionDigits or significantDigits, indicating which rounding strategy to use, as discussed in .
```
to
```
[[RoundingType]] is one of the values fractionDigits<del> or significantDigits</del><ins>, significantDigits, morePrecision, or  lessPrecision</ins>, indicating which rounding strategy to use, as discussed in .
```

### LINKED ISSUE #45: Addition to the "Table 3: Resolved Options of NumberFormat Instances" should be <ins> highlighted
https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/diff.html#table-numberformat-resolvedoptions-properties

Table 3: Resolved Options of NumberFormat Instances

I think you need to add &lt;ins> to the following rows

Internal Slot | Property
-- | --
[[RoundingMode]] | "roundingMode"
[[RoundingIncrement]] | "roundingIncrement"
[[TrailingZeroDisplay]] | "trailingZeroDisplay"

### LINKED ISSUE #47: Need to add roundingPriority to Table 3: Resolved Options of NumberFormat Instances
We need to add a row of roundingPriority to  https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/diff.html#table-numberformat-resolvedoptions-properties

Internal Slot | Property
-- | --
[[RoundingPriority]] | "roundingPriority"

in Table 3: Resolved Options of NumberFormat Instances

### LINKED ISSUE #49: [[RoundingIncrement]] not passed to ToRawPrecision/ToRawFixed in FormatNumericToString
`FormatNumericToString` doesn't pass `[[RoundingIncrement]]` to `ToRawPrecision` resp. `ToRawFixed`.

### LINKED ISSUE #50: Add range checks to ResolvePluralRange to match PartitionNumberRangePattern?
Should `ResolvePluralRange` perform the same range checks as `PartitionNumberRangePattern` (steps 1-4)?

### LINKED ISSUE #53: ToIntlMathematicalValue changes -0 to +0
`Number::toString(-0)` is `"0"`, see [Number::toString](https://tc39.es/ecma262/#sec-numeric-types-number-tostring).

### LINKED ISSUE #55: CollapseNumberRange references the resulting string which should be a resulting list
In the prose describing CollapseNumberRange, the last line is:

> The algorithm is implementation dependent, but it must not introduce ambiguity: the resulting **string** after modification should be unique for all pairs x and y. 

It should instead be

> The algorithm is implementation dependent, but it must not introduce ambiguity: the resulting **list** after modification should be unique for all pairs x and y. 

https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/diff.html#sec-collapsenumberrange

### LINKED ISSUE #56: PartitionNumberRangePattern return value is infallible
In 

https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/diff.html#sec-partitionnumberrangepattern

Line 15 should be infallible, that is: `Return ! CollapseNumberRange(result).`

### LINKED ISSUE #59: PartitionNotationSubPattern order of if-then-else could be flipped
In [PartitionNotationSubPattern](https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/diff.html#sec-partitionnotationsubpattern), we have on step 6 the following:

```
6. If the numberFormat.[[UseGrouping]] is not false, then

    i) ... <lots of instructions>
7) Else,
    i) Append a new Record { [[Type]]: "integer", [[Value]]: integer } as the last element of result.
```

I think this would be easier to read if, rather than going with "not false" (which is hard to understand), we flip the conditional and say:

```
6. If the numberFormat.[[UseGrouping]] is  false, then
    i)  Append a new Record { [[Type]]: "integer", [[Value]]: integer } as the last element of result.
7) Else,
    i) ... <lots of instructions>
```

### LINKED ISSUE #61: Remove "minInteger" parameter from ToRawFixed
Tracking issue for https://github.com/tc39/ecma402/issues/589

### PR DESCRIPTION
**Pull request by sffc**
Fixes #33
Fixes #40
Fixes #42
Fixes #44
Fixes #45
Fixes #46
Fixes #47
Fixes #49
Fixes #50
Fixes #53
Fixes #55
Fixes #56
Fixes #58
Fixes #59
Fixes #61

### CODE REVIEW

**Comment by anba** [comment]
8881c8ce7f5c7fd6641835b24ee1aae9f185596e:
Should `selectRange` follow the other `formatRange` functions and throw a `TypeError` if either argument is `undefined` and throw a `RangeError` if either argument is `NaN`?

**Comment by anba** [comment]
bc1ca1729c014a9c47de059cabe3f7e8f90fdcab:
Also applies to `Intl.PluralRules.prototype.resolvedOptions`, right?

File: numberformat/diff.emu [hunk 1]
```diff
@@ -1472,6 +1472,12 @@
           1. Let _v_ be the value of _nf_'s internal slot whose name is the Internal Slot value of the current row.
           1. If _v_ is not *undefined*, then
             1. Perform ! CreateDataPropertyOrThrow(_options_, _p_, _v_).
+        1. <ins class="block">If _nf_.[[RoundingType]] is ~morePrecision~, then</ins>
+          1. <ins class="block">Perform ! CreateDataPropertyOrThrow(_options_, *"roundingPriority"*, *"morePrecision"*).</ins>
+        1. <ins class="block">Else if _nf_.[[RoundingType]] is ~lessPrecision~, then</ins>
+          1. <ins class="block">Perform ! CreateDataPropertyOrThrow(_options_, *"roundingPriority"*, *"lessPrecision"*).</ins>
+        1. <ins class="block">Else,</ins>
+          1. <ins class="block">Perform ! CreateDataPropertyOrThrow(_options_, *"roundingPriority"*, *"auto"*).</ins>
```
Comment (ryzokuken) [inline numberformat/diff.emu:1488]: I don't think you need `class="block"` in this case, just `<ins></ins>` should suffice.

File: numberformat/diff.emu [hunk 2]
```diff
@@ -185,7 +185,7 @@
           1. Let _result_ be ToRawFixed(_x_, _intlObject_.[[MinimumFractionDigits]], _intlObject_.[[MaximumFractionDigits]]<ins>, _intlObj_.[[RoundingIncrement]], _unsignedRoundingMode_</ins>).
         1. Else,
           1. <del class="block">Assert: _intlObject_.[[RoundingType]] is ~compactRounding~.</del>
-          1. <del class="block">Let _result_ be ToRawPrecision(_x_, 1, 2<ins>, _unsignedRoundingMode_</ins>).</del>
+          1. <del class="block">Let _result_ be ToRawPrecision(_x_, 1, 2).</del>
```
Comment (ryzokuken) [inline numberformat/diff.emu:188]: No corresponding change to `proposed.emu`?

**Review by ryzokuken — APPROVED** [review]
Going through the commits, overall LGTM with @anba's comments addressed. Also, you can skip the `class="block"` for `<ins>` and `<del>` that just cover a single line/element.

File: numberformat/diff.emu (hunk 1 above)
Comment (sffc) [reply numberformat/diff.emu:1488]: I prefer `class="block"` for entire `<li>`s and that's what I use elsewhere in this doc.  I use `<ins>` for inline elements.

**Comment by sffc** [comment]
> [8881c8c](https://github.com/tc39/proposal-intl-numberformat-v3/commit/8881c8ce7f5c7fd6641835b24ee1aae9f185596e):
> Should `selectRange` follow the other `formatRange` functions and throw a `TypeError` if either argument is `undefined` and throw a `RangeError` if either argument is `NaN`?

Filed follow-up

> [bc1ca17](https://github.com/tc39/proposal-intl-numberformat-v3/commit/bc1ca1729c014a9c47de059cabe3f7e8f90fdcab):
> Also applies to `Intl.PluralRules.prototype.resolvedOptions`, right?

Fixed

File: numberformat/diff.emu (hunk 2 above)
Comment (sffc) [reply numberformat/diff.emu:188]: No, because it's in a deleted section.

**Pull request merged**